src/hotspot/share/runtime/crac.cpp

```diff
@@ -237,6 +246,10 @@ static int checkpoint_restore(int *shmid) {
   // TODO add sync processing
 #endif //LINUX
 
+  if (ends_with(_crengine, "criuengine")) {
```

Comment:
It's strange the VM assumes the process hierarchy as implemented by criuengine. The cleaner way would be implementing something like this in criuengine itself. Probably setting PR_SET_PDEATHSIG will be complicated, but that can be overcomed by adding the code in our criu fork, if required. Anyway, the VM should not be aware that much how the checkpoint engine behaves.